src/Sprite.ts

```diff
@@ -635,77 +753,88 @@ export class Sprite extends SpriteBase {
     }
   }
 
-  say(text) {
-    clearTimeout(this._speechBubble.timeout);
+  public say(text: string): void {
+    if (this._speechBubble?.timeout) clearTimeout(this._speechBubble.timeout);
     this._speechBubble = { text: String(text), style: "say", timeout: null };
   }
 
-  think(text) {
-    clearTimeout(this._speechBubble.timeout);
+  public think(text: string): void {
+    if (this._speechBubble?.timeout) clearTimeout(this._speechBubble.timeout);
     this._speechBubble = { text: String(text), style: "think", timeout: null };
   }
 
-  *sayAndWait(text, seconds) {
-    clearTimeout(this._speechBubble.timeout);
+  public *sayAndWait(text: string, seconds: number): Yielding<void> {
+    if (this._speechBubble?.timeout) clearTimeout(this._speechBubble.timeout);
 
+    const speechBubble: SpeechBubble = { text, style: "say", timeout: null };
     let done = false;
-    const timeout = setTimeout(() => {
-      this._speechBubble.text = "";
-      this.timeout = null;
+    const timeout = window.setTimeout(() => {
+      speechBubble.text = "";
+      speechBubble.timeout = null;
```

Comment:
Why not clear `this._speechBubble` here entirely? If we're trying to preserve an object value, why not make `_speechBubble` non-undefined and skip the `?.` optional chaining in methods above?

Reply:
After pondering this for a bit, I've come to the conclusion that this code has actually just been subtly broken the whole time. If another thread comes along and clears the speech bubble timeout, the `done` flag will never be set and we'll be stuck waiting forever.

Let's just rewrite this entirely in another PR.

Reply:
Branched to #170.